compiler/rustc_typeck/src/astconv/generics.rs

```diff
@@ -49,13 +54,39 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                 GenericArg::Type(hir::Ty { kind: hir::TyKind::Path { .. }, .. }),
                 GenericParamDefKind::Const { .. },
             ) => {
+                match inferred_arg.expect_ty().kind() {
```

Comment:
you can use the `GenericParamDef` to get the expect type, looking at the `ty::Param` argument isn't too useful